application/backend/server/src/paste/key.rs

```diff
@@ -0,0 +1,86 @@
+use crate::common::InternalServerError;
+use actix_web::http::uri;
+use awc::{http::StatusCode, Client};
+use log::info;
+use serde::Deserialize;
+use std::env;
+
+#[derive(Deserialize)]
+struct KegKeyResp {
+    key: String,
+}
+
+#[derive(Clone)]
+pub struct KeyClient {
+    scheme: uri::Scheme,
+    ip: String,
+    port: String,
+}
+
+impl KeyClient {
+    pub fn new() -> Self {
+        info!("Init KeyClient.");
+        Self {
+            scheme: env::var("KEG_SCHEME").unwrap().parse().unwrap(),
+            ip: env::var("KEG_IP").unwrap(),
+            port: env::var("KEG_PORT").unwrap(),
+        }
+    }
+
+    pub async fn get_key(&self) -> Result<String, InternalServerError> {
+        let uri = self.get_uri("/v1/key");
+        info!("Send key request to URI: {}.", uri);
+
+        let mut resp = Client::new()
```

Comment:
dont create http client for every request; should be one per thread

Reply:
Yes, it is optimal but this approach needs KeyClient class adaptation for multithreading which requires more development time. Can be added later as improvement.

Reply:
is current implementation optimal for single-threaded or multi-threaded app (and is app single or multi threaded)?

Reply:
will be discussed on next meeting.